pkg/server/license/enforcer.go

```diff
@@ -138,17 +145,34 @@ func (e *Enforcer) SetTelemetryStatusReporter(reporter TelemetryStatusReporter)
 func (e *Enforcer) Start(
 	ctx context.Context, st *cluster.Settings, db isql.DB, initialStart bool,
 ) error {
+	// We always start disabled. If an error occurs, the enforcer setup will be
+	// incomplete, but the server will continue to start. To ensure stability in
+	// that case, we leave throttling disabled.
+	e.isDisabled.Store(true)
+	startDisabled := e.getInitialIsDisabledValue()
+
 	e.maybeLogActiveOverrides(ctx)
 
+	if !startDisabled {
+		if err := e.maybeWriteClusterInitGracePeriodTS(ctx, db, initialStart); err != nil {
+			return err
+		}
+	}
+
+	// Initialize assuming there is no license. This seeds necessary values. It
+	// must be done after setting the cluster init grace period timestamp. And it
+	// is needed for testing that may be running this in isolation to the license
+	// ccl package.
+	e.RefreshForLicenseChange(ctx, LicTypeNone, time.Time{})
+
 	// Add a hook into the license setting so that we refresh our state whenever
-	// the license changes.
-	RegisterCallbackOnLicenseChange(ctx, st)
+	// the license changes. This will also update the state for the current
+	// license if not in test.
+	RegisterCallbackOnLicenseChange(ctx, st, e)
```

Comment:
This method is also calling `RefreshForLicenseChange` and in line 166 also, we are calling the same method on the same enforcer. Any reason for this ?